build_tools/benchmarks/common/common_arguments.py

```diff
@@ -155,17 +155,33 @@ def __init__(self, *args, **kwargs):
         "for). In that case, no --benchmark_repetitions flag will be passed."
         " If not specified, a --benchmark_repetitions will be passed "
         "instead.")
-    self.add_argument("--run_config",
+    self.add_argument("--execution_benchmark_config",
                       type=_check_file_path,
                       default=None,
-                      help="JSON file of the run config")
+                      help="JSON config for the execution benchmarks")
+    self.add_argument("--target_device_name",
+                      type=str,
+                      default=None,
+                      help="Target device in benchmark config to run")
 
   def parse_args(
       self, arg_strs: Optional[Sequence[str]] = None) -> argparse.Namespace:
     args = super().parse_args(arg_strs)
 
-    if args.e2e_test_artifacts_dir is not None and args.run_config is None:
-      raise self.error("--e2e_test_artifacts_dir requires --run_config.")
+    # TODO(#11076): Remove these checks and make --execution_benchmark_config
+    # and --target_device_name required args.
+    use_new_benchmark_suite = (args.execution_benchmark_config is not None or
+                               args.target_device_name is not None)
+    if use_new_benchmark_suite:
+      if (args.execution_benchmark_config is None or
+          args.target_device_name is None):
+        self.error(
+            "--execution_benchmark_config and --target_device_name must be set together."
+        )
+    elif args.e2e_test_artifacts_dir is not None:
+      self.error(
+          "--e2e_test_artifacts_dir requires --execution_benchmark_config and --target_device_name."
+      )
```

Comment:
Probably should use subparser here to enforce these requirements. But subparser doesn't support a default option (always need to specify `legacy`, `new`). I prefer not to touch the legacy way to run the tool in this change.

Reply:
I think you could add a subparser that isn't required and leave all the legacy arguments in the main parser. I think that may be more fiddly than it's worth though

Reply:
Yeah, also I don't think it will work with the trick that putting all common arguments behind the subparser option (it requires to move all common options into the parent parser of the subparsers).

In 2~3 weeks we will clearly make the current options as legacy. At that point we can have subparser for legacy and new one.